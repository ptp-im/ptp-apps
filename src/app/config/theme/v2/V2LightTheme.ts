import color from 'color';
import { MD2LightTheme } from 'react-native-paper';
import type { MD2Theme } from 'react-native-paper';

import { black, red800, white } from './colors';

export const V2LightTheme: MD2Theme = {
  ...MD2LightTheme,
  dark: false,
  roundness: 4,
  version: 2,
  isV3: false,
  colors: {
    ...MD2LightTheme.colors,
    // primary: '#6200ee',
    accent: '#03dac4',
    // background: '#f6f6f6',
    // background: grey50,
    surface: white,
    error: '#B00020',
    // text: black,
    // onSurface: '#000000',
    disabled: color(black).alpha(0.26).rgb().string(),
    placeholder: color(black).alpha(0.54).rgb().string(),
    backdrop: color(black).alpha(0.5).rgb().string(),
    // notification: pinkA400,

    msgUnreadBadge: red800,
    onSurface: 'rgb(30, 107, 192)',
    primary: 'rgb(30, 107, 192)',
    background: 'rgb(235, 235, 240)',
    white: 'rgb(255, 255, 255)',
    background1: 'rgb(235, 235, 240)',
    backgroundSelected: 'rgb(62, 125, 185)',
    backgroundTabs: 'rgb(255, 255, 255)',
    card: 'rgb(255, 255, 255)',
    text: 'rgb(28, 28, 30)',
    textPrimary: 'rgb(10, 132, 255)',
    textAlert: 'rgb(255, 59, 48)',
    border: 'rgb(229, 229, 229)',
    notification: 'rgb(255, 59, 48)',
  },
};

export const V2NavLightColors = {
  primary: V2LightTheme.colors.primary,
  background: V2LightTheme.colors.background,
  card: V2LightTheme.colors.card,
  text: V2LightTheme.colors.text,
  border: V2LightTheme.colors.border,
  notification: V2LightTheme.colors.notification,
};
