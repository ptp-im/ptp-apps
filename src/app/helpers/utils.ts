export function getUserIndexFromLocationUrl() {
  const location = window.location;
  const url = new URL(location.href);
  if (!url.searchParams.get('u') || !Number(url.searchParams.get('u'))) {
    return 0;
  } else {
    return Number(url.searchParams.get('u'));
  }
}

export function maskAddress(address: string) {
  return `${address.substring(0, 3)}**${address.substring(address.length - 3)}`;
}
