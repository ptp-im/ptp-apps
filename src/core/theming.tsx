import type { ColorValue } from 'react-native';

import { createTheming } from '@callstack/react-theme-provider';
import {
  argbFromHex,
  themeFromSourceColor,
} from '@material/material-color-utilities';
import color from 'color';

import {
  MD2DarkTheme,
  MD2LightTheme,
  MD3DarkTheme,
  MD3LightTheme,
} from '../styles/themes';
import { tokens } from '../styles/themes/v3/tokens';
import type {
  Theme,
  MD2Theme,
  MD3Theme,
  MD3Colors,
  MD3AndroidColors,
} from '../types';

export const DefaultTheme = MD3LightTheme;

const {
  ThemeProvider,
  withTheme,
  useTheme: useThemeProviderTheme,
} = createTheming<Theme>(DefaultTheme);

const useTheme = (overrides?: Parameters<typeof useThemeProviderTheme>[0]) =>
  useThemeProviderTheme<MD2Theme | MD3Theme>(overrides);

export { ThemeProvider, withTheme, useTheme };

export const defaultThemesByVersion = {
  2: {
    light: MD2LightTheme,
    dark: MD2DarkTheme,
  },
  3: {
    light: MD3LightTheme,
    dark: MD3DarkTheme,
  },
};

export const getTheme = (isDark = false, isV3 = true) => {
  const themeVersion = isV3 ? 3 : 2;
  const scheme = isDark ? 'dark' : 'light';

  return defaultThemesByVersion[themeVersion][scheme];
};

type Config = {
  sourceColor: ColorValue;
};

type Schemes = {
  lightScheme: MD3Colors;
  darkScheme: MD3Colors;
};

export const getDynamicThemeElevations = (scheme: MD3AndroidColors) => {
  const elevationValues = ['transparent', 0.05, 0.08, 0.11, 0.12, 0.14];
  return elevationValues.reduce((elevations, elevationValue, index) => {
    return {
      ...elevations,
      [`level${index}`]:
        index === 0
          ? elevationValue
          : color(scheme.surface)
              .mix(color(scheme.primary), elevationValue as number)
              .rgb()
              .string(),
    };
  }, {});
};

export const createDynamicThemeColors = ({ sourceColor }: Config): Schemes => {
  const { opacity } = tokens.md.ref;
  const modes = ['light', 'dark'] as const;

  const { schemes, palettes } = themeFromSourceColor(
    argbFromHex(color(sourceColor).hex())
  );

  const { light, dark } = modes.reduce(
    (prev, curr) => {
      const schemeModeJSON = schemes[curr].toJSON();

      const customColors = {
        surfaceDisabled: color(schemeModeJSON.onSurface)
          .alpha(opacity.level2)
          .rgb()
          .string(),
        onSurfaceDisabled: color(schemeModeJSON.onSurface)
          .alpha(opacity.level4)
          .rgb()
          .string(),
        backdrop: color(palettes.neutralVariant.tone(20))
          .alpha(0.4)
          .rgb()
          .string(),
      };

      const elevation = getDynamicThemeElevations(schemeModeJSON);

      const dynamicThemeColors = Object.assign(
        {},
        ...Object.entries(schemeModeJSON).map(([colorName, colorValue]) => ({
          [colorName]: color(colorValue).rgb().string(),
        })),
        {
          elevation,
          ...customColors,
        }
      );

      return {
        ...prev,
        [curr]: dynamicThemeColors,
      };
    },
    { light: MD3LightTheme.colors, dark: MD3DarkTheme.colors }
  );

  return { lightScheme: light, darkScheme: dark };
};
