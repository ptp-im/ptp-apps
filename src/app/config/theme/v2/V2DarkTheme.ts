import color from 'color';

import type { MD2Theme } from 'react-native-paper';
import { black, red800, white } from './colors';
import { V2LightTheme } from './V2LightTheme';

export const V2DarkTheme: MD2Theme = {
  ...V2LightTheme,
  dark: true,
  mode: 'adaptive',
  version: 2,
  isV3: false,
  colors: {
    ...V2LightTheme.colors,
    // primary: '#BB86FC',
    accent: '#03dac6',
    // background: '#121212',
    surface: '#121212',
    error: '#CF6679',
    onSurface: '#FFFFFF',
    // text: white,
    disabled: color(white).alpha(0.38).rgb().string(),
    placeholder: color(white).alpha(0.54).rgb().string(),
    backdrop: color(black).alpha(0.5).rgb().string(),
    // notification: pinkA100,

    msgUnreadBadge: red800,
    primary: 'rgb(10, 132, 255)',
    background: 'rgb(1, 1, 1)',
    card: 'rgb(18, 18, 18)',
    text: 'rgb(229, 229, 231)',
    textPrimary: 'rgb(10, 132, 255)',
    border: 'rgb(39, 39, 41)',
    notification: 'rgb(255, 69, 58)',
  },
};

export const V2NavDarkColors = {
  primary: V2LightTheme.colors.primary,
  background: V2LightTheme.colors.background,
  card: V2LightTheme.colors.card,
  text: V2LightTheme.colors.text,
  border: V2LightTheme.colors.border,
  notification: V2LightTheme.colors.notification,
};
