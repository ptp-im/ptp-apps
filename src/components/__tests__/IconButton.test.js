import * as React from 'react';

import color from 'color';
import renderer from 'react-test-renderer';

import { getTheme } from '../../core/theming';
import { pink500 } from '../../styles/themes/v2/colors';
import IconButton from '../IconButton/IconButton.tsx';
import { getIconButtonColor } from '../IconButton/utils';

it('renders icon button by default', () => {
  const tree = renderer.create(<IconButton icon='camera' />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders icon button with color', () => {
  const tree = renderer
    .create(<IconButton icon='camera' iconColor={pink500} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders icon button with size', () => {
  const tree = renderer.create(<IconButton icon='camera' size={30} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders disabled icon button', () => {
  const tree = renderer.create(<IconButton icon='camera' disabled />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders icon change animated', () => {
  const tree = renderer.create(<IconButton icon='camera' animated />).toJSON();

  expect(tree).toMatchSnapshot();
});

describe('getIconButtonColor - icon color', () => {
  it('should return custom icon color', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
        customIconColor: 'purple',
      })
    ).toMatchObject({
      iconColor: 'purple',
    });
  });

  it('should return correct disabled color, for theme version 3', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
        disabled: true,
      })
    ).toMatchObject({
      iconColor: getTheme().colors.onSurfaceDisabled,
    });
  });

  it('should return theme icon color, for theme version 3, mode contained', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
        mode: 'contained',
      })
    ).toMatchObject({
      iconColor: getTheme().colors.primary,
    });
  });

  it('should return theme icon color, for theme version 3, mode contained, selected', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
        mode: 'contained',
        selected: true,
      })
    ).toMatchObject({
      iconColor: getTheme().colors.onPrimary,
    });
  });

  it('should return theme icon color, for theme version 3, mode contained-tonal', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
        mode: 'contained-tonal',
      })
    ).toMatchObject({
      iconColor: getTheme().colors.onSurfaceVariant,
    });
  });

  it('should return theme icon color, for theme version 3, mode contained-tonal, selected', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
        mode: 'contained-tonal',
        selected: true,
      })
    ).toMatchObject({
      iconColor: getTheme().colors.onSecondaryContainer,
    });
  });

  it('should return theme icon color, for theme version 3, mode outlined', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
        mode: 'outlined',
      })
    ).toMatchObject({
      iconColor: getTheme().colors.onSurfaceVariant,
    });
  });

  it('should return theme icon color, for theme version 3, mode outlined, selected', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
        mode: 'outlined',
        selected: true,
      })
    ).toMatchObject({
      iconColor: getTheme().colors.inverseOnSurface,
    });
  });

  it('should return theme icon color, for theme version 3', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
      })
    ).toMatchObject({
      iconColor: getTheme().colors.onSurfaceVariant,
    });
  });

  it('should return theme icon color, for theme version 3, selected', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
        selected: true,
      })
    ).toMatchObject({
      iconColor: getTheme().colors.primary,
    });
  });

  it('should return theme icon color, for theme version 2', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(false, false),
      })
    ).toMatchObject({
      iconColor: getTheme(false, false).colors.text,
    });
  });
});

describe('getIconButtonColor - background color', () => {
  it('should return custom background color', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
        customContainerColor: 'purple',
      })
    ).toMatchObject({
      backgroundColor: 'purple',
    });
  });

  ['contained', 'contained-tonal'].forEach((mode) =>
    it(`should return correct disabled color, for theme version 3, ${mode} mode`, () => {
      expect(
        getIconButtonColor({
          theme: getTheme(),
          mode,
          disabled: true,
        })
      ).toMatchObject({ backgroundColor: getTheme().colors.surfaceDisabled });
    })
  );

  it('should return theme icon color, for theme version 3, mode contained', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
        mode: 'contained',
      })
    ).toMatchObject({
      backgroundColor: getTheme().colors.surfaceVariant,
    });
  });

  it('should return theme icon color, for theme version 3, mode contained, selected', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
        mode: 'contained',
        selected: true,
      })
    ).toMatchObject({
      backgroundColor: getTheme().colors.primary,
    });
  });

  it('should return theme icon color, for theme version 3, mode contained-tonal', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
        mode: 'contained-tonal',
      })
    ).toMatchObject({
      backgroundColor: getTheme().colors.surfaceVariant,
    });
  });

  it('should return theme icon color, for theme version 3, mode contained-tonal, selected', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
        mode: 'contained-tonal',
        selected: true,
      })
    ).toMatchObject({
      backgroundColor: getTheme().colors.secondaryContainer,
    });
  });

  it('should return theme icon color, for theme version 3, mode outlined, selected', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
        mode: 'outlined',
        selected: true,
      })
    ).toMatchObject({
      backgroundColor: getTheme().colors.inverseSurface,
    });
  });

  it('should return undefined, for theme version 3, if mode not specified', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
      })
    ).toMatchObject({
      backgroundColor: undefined,
    });
  });
});

describe('getIconButtonColor - border color', () => {
  it('should return correct disabled color, for theme version 3', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
        disabled: true,
      })
    ).toMatchObject({
      borderColor: getTheme().colors.surfaceDisabled,
    });
  });

  it('should return theme color, for theme version 3', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
      })
    ).toMatchObject({
      borderColor: getTheme().colors.outline,
    });
  });

  it('should return undefined, for theme version 2', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(false, false),
      })
    ).toMatchObject({
      borderColor: undefined,
    });
  });
});

describe('getIconButtonColor - ripple color', () => {
  it('should return theme color, for theme version 3', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(),
      })
    ).toMatchObject({
      rippleColor: color(getTheme().colors.onSurfaceVariant)
        .alpha(0.12)
        .rgb()
        .string(),
    });
  });

  it('should return theme color, for theme version 2', () => {
    expect(
      getIconButtonColor({
        theme: getTheme(false, false),
      })
    ).toMatchObject({
      rippleColor: color(getTheme(false, false).colors.text)
        .alpha(0.32)
        .rgb()
        .string(),
    });
  });
});
