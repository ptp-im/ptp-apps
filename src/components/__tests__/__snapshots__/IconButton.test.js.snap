// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders disabled icon button 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignSelf": undefined,
      "bottom": undefined,
      "left": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
      "top": undefined,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": undefined,
          "borderColor": "rgba(28, 27, 31, 0.12)",
          "borderRadius": 20,
          "borderWidth": 0,
          "elevation": 0,
          "height": 40,
          "margin": 6,
          "overflow": "hidden",
          "width": 40,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessibilityTraits={
          Array [
            "button",
            "disabled",
          ]
        }
        accessible={true}
        centered={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            false,
            Object {
              "alignItems": "center",
              "flexGrow": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgba(28, 27, 31, 0.38)",
                "fontSize": 24,
              },
            ]
          }
        >
          □
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders icon button by default 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignSelf": undefined,
      "bottom": undefined,
      "left": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
      "top": undefined,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": undefined,
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 20,
          "borderWidth": 0,
          "elevation": 0,
          "height": 40,
          "margin": 6,
          "overflow": "hidden",
          "width": 40,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessibilityTraits="button"
        accessible={true}
        centered={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            false,
            Object {
              "alignItems": "center",
              "flexGrow": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgba(73, 69, 79, 1)",
                "fontSize": 24,
              },
            ]
          }
        >
          □
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders icon button with color 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignSelf": undefined,
      "bottom": undefined,
      "left": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
      "top": undefined,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": undefined,
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 20,
          "borderWidth": 0,
          "elevation": 0,
          "height": 40,
          "margin": 6,
          "overflow": "hidden",
          "width": 40,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessibilityTraits="button"
        accessible={true}
        centered={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            false,
            Object {
              "alignItems": "center",
              "flexGrow": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "#e91e63",
                "fontSize": 24,
              },
            ]
          }
        >
          □
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders icon button with size 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignSelf": undefined,
      "bottom": undefined,
      "left": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
      "top": undefined,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": undefined,
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 23,
          "borderWidth": 0,
          "elevation": 0,
          "height": 46,
          "margin": 6,
          "overflow": "hidden",
          "width": 46,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessibilityTraits="button"
        accessible={true}
        centered={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            false,
            Object {
              "alignItems": "center",
              "flexGrow": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgba(73, 69, 79, 1)",
                "fontSize": 30,
              },
            ]
          }
        >
          □
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders icon change animated 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignSelf": undefined,
      "bottom": undefined,
      "left": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
      "top": undefined,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": undefined,
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 20,
          "borderWidth": 0,
          "elevation": 0,
          "height": 40,
          "margin": 6,
          "overflow": "hidden",
          "width": 40,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessibilityTraits="button"
        accessible={true}
        centered={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            false,
            Object {
              "alignItems": "center",
              "flexGrow": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 24,
                "width": 24,
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              }
            }
          >
            <Text
              accessibilityElementsHidden={true}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "color": "rgba(73, 69, 79, 1)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              □
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
